src/qtgui/plotter.cpp

```diff
@@ -1252,7 +1253,8 @@ void CPlotter::draw(bool newData)
     const double frameTime = 1.0 / (double)fft_rate;
 
     // Redraw the plot if it is visible.
-    const bool doPlotter = !m_2DPixmap.isNull();
+    const bool doPlotter = (!m_2DPixmap.isNull()
+        && tnow_ms >= tlast_plot_drawn_ms + PLOTTER_UPDATE_LIMIT_MS);
```

Comment:
One quirk of this calculation method is that 30 fps gives a refresh rate of 30 Hz, whereas the next step (40 fps) gives 20 Hz, which is noticeably slower. I'll have to think about whether there's a better way.

Reply:
There's also enough jitter in `tnow_ms` that some frames are often skipped even at 30 fps.

Reply:
We may want to think about what "frame rate" means for the app. Some stuff can be completely automatic and some stuff matters to the user.

Reply:
Agreed. I wonder whether it might be possible to drive things with vsync, so that drawing occurs (at most) once per frame. Animation might be smoother that way.